### ISSUE TITLE
Running docker compose by stack in portainer but can't connect to the controller

### ISSUE THREAD

**Issue by luvxinh** [mbentley/docker-omada-controller#180]
**Describe the bug**
The container is up and running in Portainer but can't connect to the controller

**To Reproduce**

Browser: Firefox latest as at the date of writing this.
openmediavault v6. Storage is created on an external hdd. File system ext4.
Portainer latest installed by openmediavault.
Docker latest installed by openmediavault.
Docker swarm including 2 nodes of Raspberry pi 4.

I use the docker compose file to deploy a stack on Portainer. It spins up and the controller started, but I can't connect to it either on port 8088 or 8043.

**Expected behavior**
Should be connect-able.

**How you're launching your container**
copy and paste the below in the stack deployment of Portainer.

```
version: "3.1"

services:
   omada-controller:
     container_name: omada-controller
     image: mbentley/omada-controller:latest
     environment:
      - TZ=Etc/UTC
      - MANAGE_HTTP_PORT=8088
      - MANAGE_HTTPS_PORT=8043
      - PORTAL_HTTP_PORT=8088
      - PORTAL_HTTPS_PORT=8043
      - SHOW_SERVER_LOGS=true
      - SHOW_MONGODB_LOGS=false
      - SSL_CERT_NAME="tls.crt"
      - SSL_KEY_NAME="tls.key"
     network_mode: host

     volumes:
      - /srv/dev-disk-by-uuid-6dafc5d3-83f5-5c47-acbc-d30349bf2aee/omada/data:/opt/tplink/EAPController/data
      - /srv/dev-disk-by-uuid-6dafc5d3-83f5-5c47-acbc-d30349bf2aee/omada/work:/opt/tplink/EAPController/work
      - /srv/dev-disk-by-uuid-6dafc5d3-83f5-5c47-acbc-d30349bf2aee/omada/logs:/opt/tplink/EAPController/logs
      
restart: unless-stopped
#volumes:
#  omada-data:
#  omada-work:
#  omada-logs:

```
**Container Logs**
```
INFO: Time zone set to 'Etc/UTC'
INFO: Setting 'portal.https.port' to 8043 in omada.properties
WARNING: owner or group (0:100) not set correctly on '/opt/tplink/EAPController/data'
INFO: setting correct permissions
WARNING: owner or group (0:100) not set correctly on '/opt/tplink/EAPController/work'
INFO: setting correct permissions
WARNING: owner or group (0:100) not set correctly on '/opt/tplink/EAPController/logs'
INFO: setting correct permissions
INFO: Database directory missing; creating '/opt/tplink/EAPController/data/db'
done
INFO: Starting Omada Controller as user omada
tail: cannot open '/opt/tplink/EAPController/logs/server.log' for reading: No such file or directory
tail: '/opt/tplink/EAPController/logs/server.log' has appeared;  following new file
02-12-2022 12:33:49.645 INFO [log4j-thread] [] c.t.s.o.c.o.a.b(32): success to load configuration omada.properties
02-12-2022 12:33:49.777 INFO [main] [] c.t.s.o.s.OmadaBootstrap(79): going to start local mongod.
02-12-2022 12:33:53.811 INFO [main] [] c.t.s.o.s.s.b(161): mongodb process id is 51
02-12-2022 12:33:53.816 ERROR [main] [] c.t.s.f.c.FacadeUtils(68): facadeMsgEnable is not enable, msg: Mongo DB server started
02-12-2022 12:33:53.819 INFO [main] [] c.t.s.o.s.s.b(142): Mongo DB server started

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v2.4.3)

02-12-2022 12:33:58.102 INFO [main] [] c.t.s.o.s.OmadaLinuxMain(55): Starting OmadaLinuxMain v5.0.30 using Java 1.8.0_312 on da89444ff115 with PID 1 (/opt/tplink/EAPController/lib/local-starter-5.0.30.jar started by omada in /opt/tplink/EAPController/lib)
02-12-2022 12:33:58.114 INFO [main] [] c.t.s.o.s.OmadaLinuxMain(660): No active profile set, falling back to default profiles: default
02-12-2022 12:34:31.379 WARN [main] [] o.s.d.c.CustomConversions(260): Registering converter from class java.time.LocalDateTime to class org.joda.time.LocalDateTime as reading converter although it doesn't convert from a store-supported type! You might want to check your annotation setup at the converter implementation.
02-12-2022 12:34:40.210 WARN [main] [] o.s.d.c.CustomConversions(260): Registering converter from class java.time.LocalDateTime to class org.joda.time.LocalDateTime as reading converter although it doesn't convert from a store-supported type! You might want to check your annotation setup at the converter implementation.
02-12-2022 12:35:59.520 INFO [main] [] c.t.s.l.c.s.d.CacheLogConsumeHandler(61): log mq consume task is start...
02-12-2022 12:35:59.538 INFO [main] [] c.t.s.o.s.OmadaLinuxMain(61): Started OmadaLinuxMain in 123.803 seconds (JVM running for 134.686)
02-12-2022 12:36:00.489 INFO [main] [] c.t.s.o.s.p.m.d.b(88): No startUpInfo exists.
02-12-2022 12:36:01.370 INFO [main] [] c.t.s.o.s.t.OmadacInitTask(76): Start init data.
02-12-2022 12:36:01.372 ERROR [main] [] c.t.s.f.c.FacadeUtils(68): facadeMsgEnable is not enable, msg: Start init data.
02-12-2022 12:36:01.549 WARN [https-jsse-nio-8043-exec-8] [] o.s.d.c.CustomConversions(260): Registering converter from class java.time.LocalDateTime to class org.joda.time.LocalDateTime as reading converter although it doesn't convert from a store-supported type! You might want to check your annotation setup at the converter implementation.
02-12-2022 12:36:01.618 WARN [https-jsse-nio-8043-exec-8] [] o.s.d.c.CustomConversions(260): Registering converter from class java.time.LocalDateTime to class org.joda.time.LocalDateTime as reading converter although it doesn't convert from a store-supported type! You might want to check your annotation setup at the converter implementation.
02-12-2022 12:36:04.076 INFO [main] [] c.t.s.o.i.a.b.d(76): accountId=Default tenant is not exist.
02-12-2022 12:36:16.895 INFO [main] [] c.t.s.o.m.c.d.m.s.a(149): Site SiteId(6207a9b9f9ba3918a692aad2) create success in omadac OmadacId(508486138e337dafa48ddb5a358e2647).
02-12-2022 12:36:18.643 INFO [main] [] c.t.s.o.s.t.OmadacInitTask(98): succeed in initializing default omadac OmadacVO(id=508486138e337dafa48ddb5a358e2647, name=Omada Controller_8E2647)
02-12-2022 12:36:22.087 INFO [main] [] c.t.s.o.m.d.p.t.TransportConfiguration(97): manager workGroup core thread  num is 16, max thread num is 16
02-12-2022 12:36:27.258 INFO [main] [] c.t.s.e.s.c.c(152): start schedule remove expire device... period = 10
02-12-2022 12:36:27.286 INFO [main] [] c.t.s.e.s.c.c(145): update rateLimiterCache, permitsPerSecond = 5.0
02-12-2022 12:36:27.540 INFO [main] [] c.t.s.o.m.d.p.t.TransportConfiguration(124): upgrade workGroup core thread num is 8, max thread num is 8
02-12-2022 12:36:27.632 INFO [main] [] c.t.s.o.m.d.p.t.TransportConfiguration(151): upgradeSendReq workGroup core thread num is 8, max thread num is 8
02-12-2022 12:36:30.095 INFO [main] [] c.t.s.o.m.d.p.t.TransportConfiguration(70): adopt workGroup core thread num is 8, max thread num is8
02-12-2022 12:36:31.773 INFO [main] [] c.t.s.o.m.d.p.t.TransportConfiguration(43): discovery workGroup core thread num is 2, max thread num is 10
02-12-2022 12:36:38.283 INFO [main] [] c.t.s.o.p.b.a.c(72): DisconnectRequestServer start
02-12-2022 12:36:39.382 INFO [main] [] c.t.s.o.d.g.a(124): discovery Port is 29810.
02-12-2022 12:36:39.383 INFO [main] [] c.t.s.o.d.g.a(125): manage    port v1 is 29811.
02-12-2022 12:36:39.384 INFO [main] [] c.t.s.o.d.g.a(126): manage    port v2 is 29814.
02-12-2022 12:36:39.384 INFO [main] [] c.t.s.o.d.g.a(127): adopt     port v1 is 29812.
02-12-2022 12:36:39.384 INFO [main] [] c.t.s.o.d.g.a(128): upgrade   port is 29813.
02-12-2022 12:36:39.578 INFO [main] [] c.t.s.e.t.a.t.AbstractServer(53): Start NettyUdpServer bind /0.0.0.0:29810, export localhost/127.0.0.1:29810
02-12-2022 12:36:39.611 INFO [main] [] c.t.s.e.t.n.NettyTcpServer(119): TCP server /0.0.0.0:29812 global traffic shaping, writeLimit: 477364224, readLimit: 0, checkInterval: 1000, max wait time: 15000
02-12-2022 12:36:39.642 INFO [main] [] c.t.s.e.t.a.t.AbstractServer(53): Start NettyTcpServer bind /0.0.0.0:29812, export localhost/127.0.0.1:29812
02-12-2022 12:36:39.645 INFO [main] [] c.t.s.e.t.n.NettyTcpServer(119): TCP server /0.0.0.0:29811 global traffic shaping, writeLimit: 477364224, readLimit: 0, checkInterval: 1000, max wait time: 15000
02-12-2022 12:36:39.650 INFO [main] [] c.t.s.e.t.a.t.AbstractServer(53): Start NettyTcpServer bind /0.0.0.0:29811, export localhost/127.0.0.1:29811
02-12-2022 12:36:39.655 INFO [main] [] c.t.s.e.t.n.NettyTcpServer(119): TCP server /0.0.0.0:29813 global traffic shaping, writeLimit: 477364224, readLimit: 0, checkInterval: 1000, max wait time: 15000
02-12-2022 12:36:39.658 INFO [main] [] c.t.s.e.t.a.t.AbstractServer(53): Start NettyTcpServer bind /0.0.0.0:29813, export localhost/127.0.0.1:29813
02-12-2022 12:36:39.676 INFO [main] [] c.t.s.e.t.n.NettyTcpServer(119): TCP server /0.0.0.0:29814 global traffic shaping, writeLimit: 477364224, readLimit: 0, checkInterval: 1000, max wait time: 15000
02-12-2022 12:36:39.680 INFO [main] [] c.t.s.e.t.a.t.AbstractServer(53): Start NettyTcpServer bind /0.0.0.0:29814, export localhost/127.0.0.1:29814
02-12-2022 12:36:39.681 ERROR [main] [] c.t.s.f.c.FacadeUtils(68): facadeMsgEnable is not enable, msg: Device connector server started.
02-12-2022 12:36:39.682 ERROR [main] [] c.t.s.f.c.FacadeUtils(68): facadeMsgEnable is not enable, msg: Omada Controller started
02-12-2022 12:36:39.683 INFO [main] [] c.t.s.o.s.t.SuccessExitTask(23): Omada Controller started
02-12-2022 12:46:27.057 INFO [vpn-certificate-clean-workgroup-0] [] c.t.s.o.m.c.d.m.s.g.a.a(98): VpnCertificateClearTask handling omadacId: OmadacId(508486138e337dafa48ddb5a358e2647)
```

**Additional context**
1. It seems like it cannot connect to such ports.

![image](https://user-images.githubusercontent.com/33537323/153713510-b1023320-43d0-493b-8276-017688357f59.png)

2. Although the network-mode in the stack file is "host", Portainer created this overlay network. I'm not sure if it's related.

![image](https://user-images.githubusercontent.com/33537323/153713719-56ef37ca-d420-4008-b994-5d1b90b21a4c.png)

3. Container up and running.

![image](https://user-images.githubusercontent.com/33537323/153715889-e8898ef8-69b7-4772-8352-d6283552c767.png)

4. I wanted to let the container to run on the manager node only, but I couldn't figure out how to put it in the docker compose file, so I manually set it in the Portainer UI. I guess this is not related but just in case.

![image](https://user-images.githubusercontent.com/33537323/153716008-a7d0045e-8920-4533-b29d-dd59d6b07c13.png)

**Comment by mbentley** [comment]
So last time I checked, you [can't use network_mode: "host" with Swarm mode](https://docs.docker.com/compose/compose-file/compose-file-v3/#network_mode).  It's been a minute since I've deployed anything to Swarm so I forget some of the requirements for mapping ports but you will probably need to statically set ports to expose if you really want to use Swarm mode + Compose.  Make sure that your controller doesn't get scheduled on a node where the persistent storage isn't available as well otherwise it'll start up like a brand new controller as nothing is pinning the container to a node.
